core/control-panel/impl/src/models/registry.rs

```diff
@@ -0,0 +1,79 @@
+use super::artifact::ArtifactId;
+use orbit_essentials::storable;
+use orbit_essentials::types::{Timestamp, UUID};
+use std::collections::BTreeMap;
+
+/// When adding tags to registry entries, if the tag is in this list then it is unique across all the 
+/// entries of the same namespace and name. If the tag exists in other entries, then it will be removed from
+/// older entries.
+/// 
+/// This value is hardcoded, and it should be updated or moved to a configurable value in the future if needed.
+pub const _UNIQUE_TAGS: [&str; 2] = ["latest", "stable"];
+
+/// The registry entry id, which is a UUID.
+pub type RegistryEntryId = UUID;
+
+/// The registry entry is a record that is stored in the registry repository.
+/// 
+/// It stores entries about wasm modules and can be extended to other entry types. When adding new entry types,
+/// the `RegistryEntryValue` enum should be updated to include the new entry type.
+#[storable]
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct RegistryEntry {
+    /// The UUID that identifies the entry in the registry.
+    pub id: RegistryEntryId,
+    /// Wether or not the entry is public, if the entry is public then it is readable by anyone.
+    pub is_public: bool,
+    /// The namespace of the entry, which is used to group entries together (e.g. "@orbit").
+    pub namespace: String,
+    /// The name of the entry, which is used to identify it (e.g. station), within each namespace the
+    /// name should refer to a the same type of entry, but many entries can exist with the same name.
+    ///
+    /// e.g. if the namespace is "@orbit" and the name is "station", then all the entries
+    /// will refer to a wasm module that is a station.
+    pub name: String,
```

Comment:
Why do we need to distinguish a namespace and a name? What would be an example?

Reply:
This is mostly helpful to search for all things included in a given namespace, this way we can tell that the orbit namespace would include the station and upgrader for example, without it we would need to either have some search by prefix, or to enforce that convention in the name itself 🤔